### ISSUE TITLE
`devspace use context` not effective

### ISSUE THREAD

**Issue by shaunc** [devspace-cloud/devspace#956]
**What happened?**  

```
% devspace use context eks_ff-dev-k8test
[done] √ Successfully set kube-context to 'eks_ff-dev-k8test'
 % devspace deploy                       

[warn]   Are you using the correct kube context?
[warn]   Current kube context: 'eks_ff-dev-k8test'
[warn]   Last    kube context: 'local'

[info]   Use the '-s / --switch-context' flag to switch to the context and namespace previously used to deploy this project
[info]   Or use the '--no-warn' flag to ignore this warning
[wait] ⠴ Will continue in 10 seconds... (0s)^C
```

**What did you expect to happen instead?**  

After using "use context" it shouldn't warn me about the obsolete last context.

**How can we reproduce the bug?** (as minimally and precisely as possible)  

Presumably the above will let you recreate. NB that I have already separately switched contexts in kubectl.

**Local Environment:**  
- DevSpace Version: [use `devspace --version`]
- Operating System: windows | linux | mac
- Deployment method: helm | kubectl apply

**Kubernetes Cluster:**  
- Cloud Provider: google | aws | azure | DevSpace Cloud | DevSpace Cloud Enterprise | other
- Kubernetes Version: [use `kubectl version`]

**Anything else we need to know?**  

/kind bug

**Comment by FabianKramm** [comment]
@shaunc thanks for opening this issue! `devspace use context` is a project independent command and does not switch the projects context, thats why you get the warning. However, I understand that its a problem to switch a projects current context/namespace, because currently that is not really possible without executing `devspace dev/deploy --no-warn` once. Maybe we could add a flag to the `devspace use context` and `devspace use namespace` commands to also switch the projects current context/namespace, what do you think about this?

**Comment by shaunc** [comment]
Yes -- flag is a good idea. (I would sort of expect it as default, as I can always type `kubectl config use-context`), but either way. Thanks!

**Comment by shaunc** [comment]
Maybe it would also be appropriate to tweak the "--no-warn" message to say "... and switch the project context" (or something similar).

**Comment by LukasGentele** [comment]
Not sure if a flag is doing anything good here. What @shaunc describes is an expectation issue of the user. An optional flag is not going to change that because then the flag should be default true/enabled. A flag which is default true however is not possible because the command should work project independent as @FabianKramm said.

Additionally an optional flag is unnecessary as it does not matter if you need to know and use an additional flag during context change or later during deploy/dev (using --no-warn). Result is the same and if it is currently surprising it will also be no matter which flag we add.

One option to fix the surprising behavior issue would be if DevSpace checks if it runs inside a project scope and then prints the message "Switching kube-context for project as well" (and then uodates the cached context name). But I'm not sure if we want that.

**Comment by shaunc** [comment]
`"Switching kube-context for project as well"` ... I would like that. Other `devspace use` commands are not project independent (e.g. `devspace use profile`). If it switches contexts in a project it still switches the global context as well, so presumably the "project independent" use case is still satisfied, except for the small sliver of users who are in a project, and want to switch the global context to something else but don't want to switch the context of the project they are in for some reason. (who would that be?) Other tools that have local context (npm, for instance) always operate on the local context by preference, and require an extra flag to do something global.

**Comment by LukasGentele** [comment]
This has been fixed in v4.12.
